src/app/booking/BookNowButton.js

```diff
@@ -1,36 +1,111 @@
-import React from 'react';
+import {  useState } from 'react';
 import { useQuery, useMutation, useQueryClient } from 'react-query';
 import { useSnackbar } from 'notistack';
 import Button from '@mui/material/Button';
-import MenuItem from '@mui/material/MenuItem';
-import { createBooking, getChildren } from "../api.mjs";
+import { getDisplayedWeekModayDate } from "@/app/booking/ActivitiesCalendar";
+import { createBooking, getChildren, createBookingSet } from "../api.mjs";
 import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
 import PopupState, { bindTrigger } from 'material-ui-popup-state';
 import { getFlatErrors } from "../activities/[activityId]/edit/components/formikFields";
 import MenuChildPopup from "./MenuChildPopup";
+import WholeWeekDialog from "./WholeWeekDialog";
+import dayjs from "dayjs";
+
+export const BookNowButton = ({ activity, targetDate, isEarlyDropOffSelected, isLatePickUpSelected }) => {
+    const displayedWeekModayDate = getDisplayedWeekModayDate(dayjs(targetDate));
+    const weekDates = Array.from({ length: 5 }, (_, i) => displayedWeekModayDate.add(i, "day"));
+    const [isDialogOpen, setDialogOpen] = useState(false);
+    const [selectedChildId, setSelectedChildId] = useState(null);
+
+    const [earlyDropOffDays, setEarlyDropOffDays] = useState({
+      M: false,
+      T: false,
+      W: false,
+      Th: false,
+      F: false,
+    });
+  
+    const [latePickupDays, setLatePickupDays] = useState({
+      M: false,
+      T: false,
+      W: false,
+      Th: false,
+      F: false,
+    });
+
+    const transformDaysToArray = (days) => {
+      return ['M', 'T', 'W', 'Th', 'F'].map(day => days[day]);
+    };
 
-export const BookNowButton = ({ activityId, targetDate, isEarlyDropOffSelected, isLatePickUpSelected }) => {
     const { data: children } = useQuery("children", getChildren);
     const queryClient = useQueryClient();
     const { enqueueSnackbar } = useSnackbar();
-    const mutation = useMutation((childId) => createBooking({ activity: activityId, child: childId, date: targetDate, isEarlyDropOff: isEarlyDropOffSelected, isLatePickUp: isLatePickUpSelected }), 
-    {
-      onSuccess: () => {
-        enqueueSnackbar("Booking created", { variant: "success" });
-        queryClient.invalidateQueries("bookings");
-      },
-      onError: (error) => {
-          const errorMessage = getFlatErrors(error).join(". ");
-          enqueueSnackbar(errorMessage, { variant: "error" });
-      },
-    });
+
+    const onSuccess = () => {
```

Comment:
These methods are not so big, why do not include them in mutation definition?
Otherwise it is harder to understand mutatino definition.